crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -77,6 +77,7 @@ fn infer_definition_types_cycle_recovery<'db>(
     _cycle: &salsa::Cycle,
     input: Definition<'db>,
 ) -> TypeInference<'db> {
+    tracing::trace!("infer_definition_types_cycle_recovery");
```

Comment:
We should change this to an actual sentence rather than the method name. Tracing messages are user facing (not `trace` but we might decide to make them user facing in the future).

Reply:
This function is a partially-broken temporary crutch just to get us through until we have fixpoint iteration and full deferred resolution of annotations in the necessary places; it should go away. So I'm not too worried about this particular trace message sticking around for a long time. But this is a good point.